diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0348f79..0006829 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,10 +54,13 @@ target_compile_definitions(GSL INTERFACE
 # the SYSTEM keyword suppresses warnings for users of the library
 if(GSL_STANDALONE_PROJECT)
     target_include_directories(GSL INTERFACE
-        $<BUILD_INTERFACE:
-            ${CMAKE_CURRENT_SOURCE_DIR}/include
-        >
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+        $<INSTALL_INTERFACE:include>
     )
+	install(TARGETS GSL EXPORT GSL 
+		DESTINATION lib) # ignored
+	install(EXPORT GSL FILE ms-gsl-config.cmake 
+		DESTINATION share/cmake/ms-gsl)
 else()
     target_include_directories(GSL SYSTEM INTERFACE
         $<BUILD_INTERFACE:
